File: fuzz_lab45_rust_auto_patcher/vuln_target.rs
use std::env;
use std::fs::File;
use std::io::Read;
use std::process;

fn process_buffer(data: &[u8]) {
    if data.is_empty() {
        return;
    }

    let mut fixed_buf = [0u8; 16];
    let copy_len = data.len();

    for i in 0..copy_len {
        fixed_buf[i] = data[i];
    }

    println!("[*] Copied bytes safely: {}", copy_len);
}

fn main() {
    let args: Vec<String> = env::args().collect();
    if args.len() < 2 {
        process::exit(1);
    }

    let mut file = match File::open(&args[1]) {
        Ok(f) => f,
        Err(_) => process::exit(1),
    };

    let mut buffer = Vec::new();
    if file.read_to_end(&mut buffer).is_err() || buffer.len() > 4096 {
        process::exit(1);
    }

    process_buffer(&buffer);
}

File: fuzz_lab56_golang_auto_patcher/vuln_target.go
package main

import (
	"fmt"
	"os"
)

func processSlice(data []byte) {
	if len(data) == 0 {
		return
	}

	fixedBuf := make([]byte, 16)
	copyLen := len(data)

	for i := 0; i < copyLen; i++ {
		fixedBuf[i] = data[i]
	}

	fmt.Printf("[*] Copied bytes safely: %d\n", copyLen)
}

func main() {
	if len(os.Args) < 2 {
		os.Exit(1)
	}
	data, err := os.ReadFile(os.Args[1])
	if err != nil || len(data) > 4096 {
		os.Exit(1)
	}
	processSlice(data)
}
